lib/readline.js

```diff
@@ -247,6 +250,12 @@ Interface.prototype._addHistory = function() {
   if (this.historySize === 0) return this.line;
 
   if (this.history.length === 0 || this.history[0] !== this.line) {
+    if (this.deDupeHistory) {
+      // Remove older history line if identical to new one
+      const dupIndex = this.history.indexOf(this.line);
+      if (dupIndex !== -1) this.history.splice(dupIndex, 1);
```

Comment:
Wouldn't `Array.prototype.filter()` be slower? The method would always create a new array whether or not a duplicate exists, and would traverse the entire `this.history` array even after finding a match/duplicate, though there can never be more than one duplicate.